spec/jobs/voice_otp_sender_job_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+include Features::ActiveJobHelper
+
+describe VoiceOtpSenderJob do
+  describe '.perform' do
+    it 'initiates the phone call to deliver the OTP', twilio: true do
+      TwilioService.telephony_service = FakeVoiceCall
+
+      VoiceOtpSenderJob.perform_now(
+        code: '1234',
+        phone: '555-5555',
+        otp_created_at: Time.current
+      )
+
+      calls = FakeVoiceCall.calls
+
+      code = '1234'.scan(/\d/).join(', ')
+      message = t('voice.otp_confirmation', code: code)
+      url_message = URI.escape(message)
+
+      expect(calls.size).to eq(1)
+      call = calls.first
+      expect(call.to).to eq('555-5555')
+      expect(call.url).to include(url_message)
+      expect(call.from).to match(/(\+19999999999|\+12222222222)/)
+    end
+
+    it 'does not send if the OTP code is expired' do
+      reset_job_queues
+      TwilioService.telephony_service = FakeVoiceCall
+      FakeVoiceCall.calls = []
+      otp_expiration_period = Devise.direct_otp_valid_for
+      user = create(:user, direct_otp_sent_at: otp_expiration_period.ago)
+
+      VoiceOtpSenderJob.perform_now(
+        code: '1234',
+        phone: '555-5555',
+        otp_created_at: user.direct_otp_sent_at
```

Comment:
Same comment here about using date instead of creating a user. We want to avoid using the DB when possible to keep tests fast.